projects/storefrontlib/cms-components/product/product-intro/product-intro.component.spec.ts

```diff
@@ -161,17 +179,68 @@ describe('ProductIntroComponent in product', () => {
       );
     });
 
-    it('should not display Show Reviews when no reviews available', () => {
+    it('should not display Show Reviews when no average rating', () => {
       productIntroComponent.product$ = of<Product>({
         averageRating: undefined,
-        numberOfReviews: 1,
       });
       fixture.detectChanges();
       expect(fixture.debugElement.nativeElement.innerText).not.toContain(
         'productSummary.showReviews'
       );
     });
 
+    it('should not display Show Reviews when reviews are not available', () => {
+      productIntroComponent.product$ = of<Product>({
+        averageRating: 5,
+      });
+      productIntroComponent['getReviewsComponent'] = () => null;
+
+      fixture.detectChanges();
+      expect(fixture.debugElement.nativeElement.innerText).not.toContain(
+        'productSummary.showReviews'
+      );
+    });
+
+    it('should not display Show Reviews when reviews component has been destroyed', () => {
+      const event = new ComponentDestroyEvent();
+      event.id = 'ProductReviewsTabComponent';
+
+      spyOn(eventService, 'get').and.returnValues(of(), of(event));
+
+      fixture = TestBed.createComponent(ProductIntroComponent);
+      productIntroComponent = fixture.componentInstance;
+
+      productIntroComponent.product$ = of<Product>({
+        averageRating: 5,
+      });
+      productIntroComponent['getReviewsComponent'] = () => null;
+
+      fixture.detectChanges();
+      expect(fixture.debugElement.nativeElement.innerText).not.toContain(
+        'productSummary.showReviews'
+      );
+    });
+
+    it('should display Show Reviews when reviews are available', () => {
```

Comment:
This test covers the case for `ComponentCreatedEvent`, so what do you think about changing the test name to:
```suggestion
    it('should display Show Reviews when reviews component has been created', () => {
```

... and btw. we could add another test saying `when the reviews component is available` (I mean: when `getReviewsComponent()` returns a truthy value from the beginning)